### PR TITLE
Deals with wildcarded virtual hosts for ports other than port 80

### PR DESCRIPTION
**Pull request by mbanikazemi**
Addresses issue:  #14033

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/route/route.go [hunk 1]
```diff
@@ -179,6 +179,10 @@ func buildSidecarVirtualHostsForVirtualService(
 		// This is a gross HACK. Fix me. Its a much bigger surgery though, due to the way
 		// the current code is written.
 		serviceByPort[80] = nil
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/route/route.go]: If we deal with all ports then do we still need this port 80 part?

**Comment by mbanikazemi** [comment]
@rshriram please have a look at this PR when you get a chance.

**Comment by howardjohn** [comment]
@mbanikazemi is this still wip?

**Comment by mbanikazemi** [comment]
@howardjohn I removed the WIP from the title.

File: pilot/pkg/networking/core/v1alpha3/route/route.go (hunk 1 above)
Comment (howardjohn) [reply pilot/pkg/networking/core/v1alpha3/route/route.go]: same comment

Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/route/route.go]: Not sure what's this for?

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/route/route.go]: I mean without this patch

Comment (sdake) [reply pilot/pkg/networking/core/v1alpha3/route/route.go]: @hzxuzhonghu do you mean to say you don't understand why this isn't just removed as part of this PR?

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/route/route.go]: I mean i donot understand why this was added previously. In my understanding, wildcard virtual host is to match a specific service, if a service name already exists, why generate extra wildcarded domains.

Comment (mbanikazemi) [reply pilot/pkg/networking/core/v1alpha3/route/route.go]: Please have a look at issue #14033 that triggered this patch.

Comment (howardjohn) [reply pilot/pkg/networking/core/v1alpha3/route/route.go]: even if they don't have a port 80 specified? 

I thought before we have a hack to set 80

Now we correctly set this, and only want 80 if they set match to 80?

I don't really understand this though so maybe that doesn't make sense

**Comment by howardjohn** [comment]
Is this still needed? I have an open comment

**Comment by howardjohn** [comment]
feel free to reopen

**Comment by FrimIdan** [comment]
Hi @howardjohn, why the issue was closed?
I've encounter the same issue when wanted to create a virtual service with HTTP match port other them 80. The commit here fixed my problem.

**Comment by howardjohn** [comment]
The change is not complete and @mbanikazemi has not responded in over 2 months

**Comment by FrimIdan** [comment]
This is a real issue, can you please tell me what is not complete and I will try to continue @mbanikazemi work.
I've tested it in my environment and it fixes my problem.

**Comment by howardjohn** [comment]
There should not be any special cases for port 80

**Comment by FrimIdan** [comment]
OK, So if no match port was specified then we want to use port 80 in order to not break current behavior?

**Comment by FrimIdan** [comment]
@howardjohn Hi, My issue was resolved by adding an extra service entry to match the new virtual service config.
I will not continue with this issue for now as I don't understated the original issue it was opened about.
Thanks!

**Pull request closed without merging**